pkg/validator/validator_test.go

```diff
@@ -157,6 +156,24 @@ var testCases = []struct {
 	// expressionUsesExistingLabels
 	{name: "labelsExists", validator: expressionUsesExistingLabels{}, promClient: prometheus.NewClientMock([]string{"__name__", "foo"}, 0, false, false), rule: rulefmt.Rule{Expr: `up{foo="bar"}`}, expectedErrors: 0},
 	{name: "labelsDoesNotExist", validator: expressionUsesExistingLabels{}, promClient: prometheus.NewClientMock([]string{"__name__"}, 0, false, false), rule: rulefmt.Rule{Expr: `up{foo="bar"}`}, expectedErrors: 1},
+
+	{name: "withName", validator: expressionWithNoMetricName{}, promClient: prometheus.NewClientMock(prometheus.NewSeriesResponseMock(2), 0, false, false), rule: rulefmt.Rule{Expr: `up{foo="bar"}`}, expectedErrors: 0},
```

Comment:
nit: same as above, feel free to omit the `promClient` if the validator does not use it

Reply:
depends if I can use a different Validator interface

Reply:
what should I pass here instead of the client?

Reply:
Well… if you drop the validation of the parameter in the validate function, you can pass just `nil` :grimacing: 

But It's ok if you leave it like this, I'm just suggesting to ignore some code paths we know are not used which might feel bit hacky.

Reply:
OK, updated

Reply:
true :) 
pushed now.